src/Components/HighChartsWrapper/HighChartsWrapper.tsx

```diff
@@ -0,0 +1,247 @@
+                                .sort((a, b) => a[0] - b[0]), // sort in case the timestamps are not in ascending order
+                            type: 'line', // by default, show series in line chart type
+                            color: highChartColor(idx), // by default, set color to use it for labels in legend to match series color
+                            marker: {
+                                enabled: false // by default, do not mark data points on series, only on hover
+                            },
+                            tooltip: {
+                                ...(sD.tooltipSuffix && {
+                                    // by default, append tooltip suffix if exist for series (e.g. unit and/or aggregation of series data)
+                                    valueSuffix: ` ${sD.tooltipSuffix}`
+                                })
+                            },
+                            yAxis: chartOptions?.hasMultipleAxes
+                                ? idx
+                                : undefined,
+                            dataGrouping: {
+                                approximation: (chartOptions.dataGrouping ===
+                                'min'
+                                    ? 'low'
+                                    : chartOptions.dataGrouping === 'max'
+                                    ? 'high'
+                                    : 'average') as DataGroupingApproximationValue
+                            }
+                        } as SeriesOptionsType)
+                ) || [],
+        [seriesData]
+    );
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    //side-effects
+    useEffect(() => {
+        if (!isLoading) {
+            chartComponentRef.current?.chart?.hideLoading();
+        }
+    }, [isLoading]);
+
+    const defaultYAxisProps: Highcharts.YAxisOptions = {
+        title: undefined, // by default, do not show any labels in y axis, only numeric range
+        labels: {
+            style: classNames.subComponentStyles.yAxis().label as CSSObject
+        }
+    };
+
+    const multipleYAxisProps: Array<Highcharts.YAxisOptions> = highChartSeries.map(
+        (_hcS, idx) => {
+            const isOnOppositeSide =
+                idx >= Math.floor(highChartSeries.length / 2) ? true : false; // by default, put the other half of the y-axes to the opposite side
+            return {
+                gridLineWidth: idx > 1 ? 0 : 1,
+                opposite: isOnOppositeSide,
+                title: undefined, // by default, do not show any labels in y axis, only numeric range
+                labels: {
+                    x: isOnOppositeSide ? 8 : -8, // to make label space less to make up more space for plot
+                    style: { color: highChartColor(idx) }
+                }
+            };
+        }
+    );
+
+    const deeplinkShareButtonDOMString = renderToString(
```

Comment:
since title.text prop of highchart requires string that is why I needed to convert jsx element to dom string using this to get the share link there. but as we discussed in the meeting, I will move the share into ellipsis in following PR along with title outside of the chart and ellipsis menu